src/components/EventEditView/EventEditView.tsx

```diff
@@ -0,0 +1,265 @@
+    const api = new EventsApi()
+    try {
+      const response = await api.getById(eventId)
+      
+      setEventName(response.name)
+      setEventDescription(response.description)
+      setEventDate(response.eventDate)
+      setAddress(response.address)
+      setPhoneNumber(response.phoneNumber)
+      setEventTag(response.tags)
+      setEventHeadquarter(response.headquarter)
+      setSelectedEventType(response.type)
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  const getValues = (setValue: Dispatch<SetStateAction<string>>, event:ChangeEvent<HTMLInputElement>) => {
+    return setValue(event.target.value)
+  }
+
+  const getValuesSelect = (setValue: Dispatch<SetStateAction<string>>, event:  ChangeEvent<{name?: string | undefined; value: unknown;}>) => {
+    return setValue(event.target.value as string)
+  }
+
+  const classes = useStyles()
+  const history = useHistory()
+
+  const [getName, setEventName] = useState<string>(eventName)
+  const [getType, setSelectedEventType] = useState<string>(eventType)
+  const [getDescription, setEventDescription] = useState<string>(eventDescription)
+  const [getDate, setEventDate] = useState<string>(eventDate)
+  const [getAddress, setAddress] = useState<string>(address)
+  const [getphone, setPhoneNumber] = useState<string>(phoneNumber)
+  const [getTag, setEventTag] = useState<string>(eventTag)
+  const [getHeadquarter, setEventHeadquarter] = useState<string>(headquarter)
+  const { id } = useParams<string>()
+
+  const updateEventType = (selectedEventType: string) => {
+    setSelectedEventType(selectedEventType)
+    fetchEventById(id)
+  }
+
+  const onChangeName = (event: ChangeEvent<HTMLInputElement>) => getValues(setEventName, event)
+  const onChangeDescription = (event: ChangeEvent<HTMLInputElement>) => getValues(setEventDescription, event)
+  const onChangeDate = (event: ChangeEvent<HTMLInputElement>) => getValues(setEventDate, event)
+  const onChangeAdress = (event: ChangeEvent<HTMLInputElement>) => getValues(setAddress, event)
+  const onChangePhoneNumber = (event: ChangeEvent<HTMLInputElement>) => getValues(setPhoneNumber, event)
+  const onChangeTag = (event:ChangeEvent<{ name?: string | undefined; value: unknown; }>) => getValuesSelect(setEventTag, event)
+  const handleHeadquarterChanged = (event:ChangeEvent<{ name?: string | undefined; value: unknown; }>) => getValuesSelect(setEventHeadquarter, event)
+
+  const handleCancelButton = () => {
+    history.push('/events/list')
+  }
+
+  const handleSubmitButton = async () => {
+    const api = new EventsApi()
+    try {
+      const data = await api.update(
+        id,
+        {
+          "name": getName,
+          "date": new Date(getDate),
+          "headquarter": getHeadquarter,
+          "address": getAddress,
+          "type": getType,
+          "description": getDescription,
+          "phoneNumber": getphone 
+        }
+      )
+
+      if (data.status == 200) handleCancelButton()
```

Comment:
3 things here.. 

1. Always equals check with type

```suggestion
      if (data.status === 200) handleCancelButton()
```

2. why we need to ask if its 200, if we called the update here and didnt throw an error it should call that function
3. handleCancelButton doesnt look like a good name for something that its being by non canceling code, like here you made an update everything went fine and then you call a cancel method, please change the name to be more generic, something like `redirect`

Reply:
Done